gammapy/datasets/simulate.py

```diff
@@ -86,7 +86,11 @@ def sample_sources(self, dataset):
                 temporal_model = ConstantTemporalModel()
 
             table = self._sample_coord_time(npred, temporal_model, dataset.gti)
-            table["MC_ID"] = idx + 1
+            if len(table) > 0:
+                table["MC_ID"] = idx + 1
+            else:
+                mcid = table.Column(name="MC_ID", length=len(table), dtype=int)
```

Comment:
Maybe put explicitly `len(table)` to `0` here?

Reply:
You're right. I changed to `0`.

I also added a new unit test called `test_mde_sample_weak_src`. This verifies that, when the source is too faint, the simulated events are related only to the background.